packages/cbioportal-utils/src/civic/CivicDataFetcher.ts

```diff
@@ -1,192 +1,271 @@
+            } else {
+                needToFetch = false;
+            }
+        }
+        return civicVariantSummaryMap;
     }
-    return variantMap;
-}
 
-/**
- * CIViC
- */
-export class CivicAPI {
-    /**
-     * Retrieves the gene entries for the ids given, if they are in the Civic API.
-     */
-    getCivicGenesBatch(ids: string): Promise<ICivicGeneSummary[]> {
-        return request
-            .get('https://civicdb.org/api/genes/' + ids)
-            .query({ identifier_type: 'entrez_id' })
-            .then(res => {
-                const response = res.body;
-                const result: CivicAPIGene[] =
-                    response instanceof Array ? response : [response];
-                return result.map((record: CivicAPIGene) => ({
+    async getCivicGeneSummaries(
+        hugoGeneSymbols: string[]
+    ): Promise<ICivicGeneSummary[]> {
+        let result: ICivicGeneSummary[] = [];
+        // civic genes api can return up to 100 civic genes in a single request (no limit on sending)
+        // if result contains more than 100 genes, the first 100 genes will be returned, other queries (genes after 100) will need to be sent again
+        // the next query will start from the last gene of previous query (by giving "after" parameter), which is the "endCursor" field returned in previous query
+        // set needToFetch to true to make sure the query will be sent at least for the first time
+        let needToFetch = true;
+        let after = null;
+        while (needToFetch) {
+            const civicGenes: CivicAPIGenes = await this.fetchCivicAPIGenes(
+                hugoGeneSymbols,
+                after
+            );
+            // hasNextPage is true means there are more than 100 genes in the return, and need to make another request
+            if (civicGenes.pageInfo?.hasNextPage) {
+                // Update endCursor when next page is available
+                after = civicGenes.pageInfo.endCursor;
+                needToFetch = true;
+            } else {
+                needToFetch = false;
+            }
+            const filteredCivicGenes = _.compact(civicGenes.nodes);
+            const geneSummaries = _.map(filteredCivicGenes, async record => {
+                const variants = await this.createCivicVariantSummaryMap(
+                    record.variants,
+                    record.id
+                );
+                const geneSummary: ICivicGeneSummary = {
                     id: record.id,
                     name: record.name,
                     description: record.description,
                     url: 'https://civicdb.org/genes/' + record.id + '/summary',
-                    variants: createVariantMap(record.variants),
-                }));
+                    variants,
+                };
+                return geneSummary;
             });
+            for (const geneSummary of geneSummaries) {
+                result.push(await geneSummary);
+            }
```

Comment:
Can we do this within the `_.map` callback above? Looks like we are already awaiting the result there. If not, a `Promise.all` is probably better than an additional for loop, something like this:

```suggestion
            result.push(...(await Promise.all(geneSummaries)));
```